matcher.go

```diff
@@ -57,6 +57,33 @@ func MatchN(pattern string, n int) MatcherFunc {
 	}
 }
 
+//MatchPalindrome creates a MatcherFunc that matches patterns that are palindrome
+func MatchPalindrome() MatcherFunc {
```

Comment:
what are your thoughts on sentence palindromes such as `race car`. technically the individual words are not palindromes but the overall sentence is. obviously running through every permutation of the line to pull out each word combination and test them is way too much overhead but i wonder if there's a clever way we can _somewhat_ support this use case.

Reply:
It doesn't highlight the sentences as of now.